sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/ChatRequestUserMessage.java

```diff
@@ -148,7 +145,17 @@ public static ChatRequestUserMessage fromJson(JsonReader jsonReader) throws IOEx
                 String fieldName = reader.getFieldName();
                 reader.nextToken();
                 if ("content".equals(fieldName)) {
-                    content = reader.getNullable(nonNullReader -> BinaryData.fromObject(nonNullReader.readUntyped()));
+                    if (reader.currentToken() == JsonToken.STRING) {
+                        content = BinaryData.fromString(reader.getString());
+                    } else if (reader.currentToken() == JsonToken.START_OBJECT) {
+                        content = BinaryData.fromObject(reader.readMap(JsonReader::readUntyped));
+                    } else if (reader.currentToken() == JsonToken.START_ARRAY) {
+                        content = BinaryData.fromObject(reader.readArray(JsonReader::readUntyped));
+                    } else if (reader.currentToken() == JsonToken.NULL) {
+                        content = null;
+                    } else {
+                        throw new IllegalStateException("Unexpected token: " + reader.currentToken());
```

Comment:
We should have a bit more context in the exception message. We should mention that we are trying to deserialize ChatRequestUserMessage and the "content" field had an unexpected token.